File: app/src/main/java/app/solocoin/solocoin/ui/home/AppGuideActivity.java
package app.solocoin.solocoin.ui.home;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.KeyEvent;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import app.solocoin.solocoin.R;

public class AppGuideActivity extends AppCompatActivity {
    private WebView webview;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_app_guide);
        webview=findViewById(R.id.webview);
        Intent intent =getIntent();
        String link=intent.getStringExtra("link");
        WebSettings webSettings =webview.getSettings();
        webSettings.setJavaScriptEnabled(true);
        MyWebViewClient webViewClient=new MyWebViewClient();
        webview.setWebViewClient(webViewClient);
        webview.loadUrl(link);
    }
    public class MyWebViewClient extends WebViewClient {
        @Override
        public boolean shouldOverrideUrlLoading(WebView view, String url) {
            if(url.indexOf("solocoin.app")>-1) return false;
            Intent intent=new Intent(Intent.ACTION_VIEW, Uri.parse(url));
            startActivity(intent);
            return true;
        }
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if((keyCode==KeyEvent.KEYCODE_BACK) && webview.canGoBack() ) {
            webview.goBack();
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: app/src/main/java/app/solocoin/solocoin/ui/home/AppGuideActivity.kt
package app.solocoin.solocoin.ui.home

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.KeyEvent
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import app.solocoin.solocoin.R
import kotlinx.android.synthetic.main.activity_app_guide.*
/**
 * Created by Karandeep Singh on 10/07/2020
 */
class AppGuideActivity: AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_app_guide)

        val intent = getIntent()
        val link = intent.getStringExtra("link")
        val webSettings = webview.getSettings()
        webSettings.javaScriptEnabled = true
        val webViewClient = MyWebViewClient()
        webview.setWebViewClient(webViewClient)
        webview.loadUrl(link)
    }
    inner class MyWebViewClient : WebViewClient() {
        override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
            if (url.indexOf("solocoin.app") > -1) return false
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(url))
         startActivity(intent)
            return true
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK && webview!!.canGoBack()) {
            webview!!.goBack()
            return true
        }
        return super.onKeyDown(keyCode, event)
    }
}